pkg/cache/listers.go

```diff
@@ -27,21 +27,29 @@ import (
 	"k8s.io/client-go/tools/cache"
 )
 
-// NewGenericClusterLister creates a new instance for the GenericClusterLister.
-func NewGenericClusterLister(indexer cache.Indexer, resource schema.GroupResource) *GenericClusterLister {
-	return &GenericClusterLister{
+// NewGenericClusterLister creates a new instance for the genericClusterLister.
+func NewGenericClusterLister(indexer cache.Indexer, resource schema.GroupResource) GenericClusterLister {
```

Comment:
Does this need to return the interface or can we return a `*genericClusterLister`?

Reply:
My linter yells at me for returning unexported types

Reply:
Should we export the type too?

Reply:
Then we need to find a new name for it, etc - I am ambivalent here. Can tackle it next week, will need updates in codegen & client-go as follow-ups.